cmake/external/gsl.cmake

```diff
@@ -0,0 +1,16 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+include(FetchContent)
+
+FetchContent_Declare(
+    GSL
+    GIT_REPOSITORY https://github.com/microsoft/gsl
+    GIT_TAG a3534567187d2edc428efd3f13466ff75fe5805c  # v4.0.0
+    GIT_SHALLOW ON
+    )
+
+FetchContent_MakeAvailable(GSL)
+
+set(GSL_TARGET "Microsoft.GSL::GSL")
+set(GSL_INCLUDE_DIR "$<TARGET_PROPERTY:${GSL_TARGET},INTERFACE_INCLUDE_DIRECTORIES>")
```

Comment:
>set(GSL_INCLUDE_ [](http://example.com/codeflow?start=0&length=16)

Let's not forget about natvis. Once experienced, it is hard to go back.

Reply:
A puppy love!

Yes. The official GSL repo has it in their cmake targets.

Reply:
Yep, it should be included if we link to the target:
https://github.com/microsoft/GSL/blob/a3534567187d2edc428efd3f13466ff75fe5805c/cmake/guidelineSupportLibrary.cmake#L79-L82